### PR TITLE
Pushing users towards the explicit actions via warning messages

### PR DESCRIPTION
**Pull request by johnsimons**
This adds the following warnings:
image
image

### CODE REVIEW

File: octopusdeploy/schema_deployment_action.go [hunk 1]
```diff
@@ -128,11 +131,6 @@ func getDeploymentActionSchema() *schema.Schema {
 	actionSchema, element := getActionSchema()
 	addActionTypeSchema(element)
 	addExecutionLocationSchema(element)
-	element.Schema["action_type"] = &schema.Schema{
```
Comment (johnsimons) [inline octopusdeploy/schema_deployment_action.go:131]: This doesn't seem to be required, it is added by `addActionTypeSchema `

File: octopusdeploy/schema_channel_rule.go [hunk 2]
```diff
@@ -37,8 +37,8 @@ func getChannelRuleSchema() map[string]*schema.Schema {
 		},
 		"id": getIDSchema(),
 		"tag": {
-			Optional:      true,
-			Type:          schema.TypeString,
+			Optional: true,
```
Comment (johnsimons) [inline octopusdeploy/schema_channel_rule.go:40]: better formatting

File: octopusdeploy/schema_deployment_action.go [hunk 3]
```diff
@@ -229,17 +227,18 @@ func getActionSchema() (*schema.Schema, *schema.Resource) {
 			},
 			"name": getNameSchema(true),
 			"notes": {
-				Description: "The notes associated with this deploymnt action.",
+				Description: "The notes associated with this deployment action.",
```
Comment (johnsimons) [inline octopusdeploy/schema_deployment_action.go:230]: while I was here, fix spelling

**Comment by johnsimons** [comment]
I couldn't figure out how to test warnings using the test lib, anyone?

**Review by jbristowe — APPROVED** [review]

**Comment by johnsimons** [comment]
Asked in https://discuss.hashicorp.com/t/how-to-test-warnings/34259

**Pull request merged**